improver/calibration/reliability_calibration.py

```diff
@@ -1441,3 +1443,54 @@ def process(
         calibrated_forecast.data = calibrated_forecast.data.astype("float32")
 
         return calibrated_forecast
+
+
+class EnforceConsistentProbabilities(PostProcessingPlugin):
+    """Reduces the probabilities of a forecast to make it consistent
+    with another provided forecast.
+    """
+
+    def __init__(self):
+        """
+        Initialise class for enforcing probabilities between two forecasts.
+        """
+
+    def process(self, forecast_cube: Cube, ref_forecast: Cube) -> Cube:
+        """
+        Lowers the probabilities of the forecast_cube to make it consistent with
+        the reference forecast. This is done by lowering the probabilities of the forecast cube,
+        to be equal to the reference forecast if they are higher than the reference forecast.
+        If the probability is already less than or equal to the reference forecast then the
+        probability is not altered.
+
+        Args:
+            forecast_cube:
+                A forecast cube of probabilities
+            ref_forecast:
+                A cube of probabilities that is used as an upper limit for
+                the forecast_cube probabilities. It must have the same dimensions
+                as the forecast_cube.
+
+        Returns:
+            A forecast cube of probabilities that has identical metadata to forecast_cube but
+            with reduced probabilities to make it consistent with ref_forecast
+        """
+
+        assert_spatial_coords_match([forecast_cube, ref_forecast])
+
+        # puts ref_forecast's data into a cube with the same metadata as forecast_cube so they
+        # can be combined
+        ref_forecast_metadata = forecast_cube.copy(data=ref_forecast.data)
+
+        diff = Combine(operation="-")([ref_forecast_metadata, forecast_cube])
+        diff.data = np.clip(diff.data, None, 0)
+
+        new_forecast = Combine(operation="+")([forecast_cube, diff])
+
+        if np.amin(diff.data) < -0.3:
```

Comment:
It's a slightly arbitrarily chosen number but is my best guess. I included the warning to try and get some idea if this plugin is causing a large changes to the cloud or is hiding a problem with another plugin (or some other calibration we try in the future) being applied to the cloud fields. 

In my experimenting I've found the inconsistencies are normally in the range of 0.01- 0.15 . Conceivably in extreme situations it could get up to 0.2 although I haven't seen this. I went with 0.3 as it would make sure that  we aren't getting warnings if there's nothing wrong but if we do get a warning its because the plugin is doing something extreme that needs to be looked at. 

I could definately see an arguemnt for dropping it down to 0.25 or even 0.2 if we wanted to be cautious.